### DISCUSSION TITLE
Request: Guide for Xen integration with ZFSBootMenu

### DISCUSSION THREAD

**Discussion by ghost** [zbm-dev/zfsbootmenu#658] [Questions]
For the last few days, I have been trying to add Xen integration with ZFSBootMenu. However, to date, I have been unsuccessful. I'm working with a Debian host and have encountered several challenges. I believe a comprehensive guide for Xen integration would be beneficial for the community.

A pre-built image for xen would be very helpful.

My learnings:
1. the default https://docs.zfsbootmenu.org/en/latest/guides/debian/bookworm-uefi.html#install-zfsbootmenu from source does not work. I had to boot from the thumbdrive with zfsbootmenu recovery image.
2. it turns out no matter what i do with debian, dracut will not build a bootable image. Hooking comes after. 
3. building on void is the only viable option as the pre-built binary is built on void.

So, as request, suppose my /boot contains xen-*.gz, a guide on how to compile on a void container for xen support would be very helpful.

Thank You

**Comment by classabbyamp** [comment]
previous discussion on the topic: #231

> the default [...] from source does not work

what does "does not work" mean?

**Reply by ghost** [reply]
On the precompiled image, it took me to enter password for zpool screen and then booted it.

> previous discussion on the topic: #231
> 
> > the default [...] from source does not work
> 
> what does "does not work" mean?

It means it does not produce a bootable image. When I select vmlinuz.EFI on reEFInd to boot, it shows me a screen as:


[ESCAPE] Recovery Shell
[RETURN] Retry immediately
'Unable to import pool, retrying in x second'


Pressing Esc or Enter does not do anything.

**Comment by zdykstra** [comment]
We'd welcome a generic guide for using ZFSBootMenu with Xen, but nothing specific to a single distribution.  I'm happy to review a PR that comes in with that information.

**Reply by ghost** [reply]
I have opened an issue regarding a memory error on xen I am facing at https://github.com/zbm-dev/zfsbootmenu/issues/660
If that gets resolved, I think the generic guide can be prepared from that.
